server/tso/allocator_manager.go

```diff
@@ -1134,3 +1140,32 @@ func (am *AllocatorManager) transferLocalAllocator(dcLocation string, serverID u
 func (am *AllocatorManager) nextLeaderKey(dcLocation string) string {
 	return path.Join(am.rootPath, dcLocation, "next-leader")
 }
+
+// UpdateMemberDCLocationInfo updates dc location of member by id.
+func (am *AllocatorManager) UpdateMemberDCLocationInfo(id uint64, dcLocation string) error {
+	key := am.member.GetDCLocationPath(id)
+	res, err := etcdutil.EtcdKVGet(am.member.Client(), key)
+	if err != nil {
+		return err
+	}
+	if len(res.Kvs) == 0 {
+		return fmt.Errorf("member %v does not exist or local tso disabled", id)
+	}
+
+	if err := am.checkDCLocationUpperLimit(dcLocation); err != nil {
+		return err
+	}
+
+	// Simplely overwrite dcLocationPath in etcd, leave the rest work to backgroup loop in AllocatorDaemon.
+	resp, err := kv.
+		NewSlowLogTxn(am.member.Client()).
```

Comment:
What about adding a comparison in the etcd transaction to make sure the old `dc-location` remains the same until we change it?

Reply:
Sounds reasonable.